sql27thsep/sql27th sep.txt

```diff
@@ -0,0 +1,224 @@
+
+1. Extract the middle 3 characters from the string 'ABCDEFG'.
+select SUBSTRING('abcdefg',3,3) as 'middleofstring ';
+
+2. From a table 'Employees' with a column 'FullName', write a query to extract the first name (assuming it's always the first word before a space).
+select	substring(full_name,1,charindex(' ',full_name+' ')-1) from Employees;
+
+3.Extract the first 5 characters from the string 'SQL Server 2022'.
+select SUBSTRING('SQL SERVER 2022' ,1,5);
+
+4. From a 'Products' table with a 'ProductCode' column, write a query to get the first 3 characters of each product code.
+select substring(productCode,1,3) from Products;
+
+5. Extract the last 4 characters from the string 'ABCDEFGHIJKL'.
+select right('abcdefghijkl',5);
+
+6.From an 'Orders' table with an 'OrderID' column (format: ORD-YYYY-NNNN), write a query to extract just the numeric portion at the end.
+  select right([product code],4) from products;
```

Comment:
How can we write this query generically, for example : if the ORDER ID is ORD-2023-123467 or ORD-2023-1234741.

Reply:
select substring([product code],len('ord-yyyy-')+1,len([product code])) from products;

Reply:
This will not work, if the the ORDER ID is ORD-2021333-123467. use with CHARINDEX and REVERSE

Reply:
yeah ok, for the one with yyyy part also more ,it wont work with this,i thought was asking about the nnn part only extended . 

 select substring(reverse([product code]),1,charindex('-',reverse([product code]))-1) from Products;